### PR TITLE
fix(dbt): correctly treat dbt-fusion test status pass as successful in asset checks (#33512)

### LINKED ISSUE #33512: [dagster-dbt] Fusion: dbt test with status 'pass' incorrectly reported as failed asset check
## Summary

When using dbt-fusion (dbt CLI v2.x) with dagster-dbt, dbt tests that pass are incorrectly reported as **failed** asset check evaluations in the Dagster UI.

## Root cause

In `dagster_dbt/core/dbt_cli_event.py`, `DbtFusionCliEventMessage._get_check_passed()` compares against `NodeStatus.Success` (`"success"`):

```python
def _get_check_passed(self) -> bool:
    return self._get_node_status() == NodeStatus.Success
```

However, dbt-fusion's `NodeFinished` event for tests reports `status: "pass"`, not `"success"`. Since `"pass" != "success"`, the check always evaluates to `passed=False`.

The dbt-core path (`DbtCoreCliEventMessage`) correctly uses `TestStatus.Pass` (which equals `"pass"`).

## Suggested fix

```python
def _get_check_passed(self) -> bool:
    return self._get_node_status() in (NodeStatus.Success, NodeStatus.Pass)
```

## Reproduction

1. Set up dagster-dbt with dbt-fusion (v2.0.0-preview.120)
2. Define a dbt test (e.g. Elementary `volume_anomalies`)
3. Run the dbt build via Dagster
4. Observe: dbt test passes (`status: pass` in logs), but `ASSET_CHECK_EVALUATION` reports the check as failed

## Environment

- dagster-dbt: 0.28.11
- dbt-fusion: 2.0.0-preview.120
- dagster: 1.12.11

### PR DESCRIPTION
**Pull request by vidiyala99**
## Summary & Motivation
Fixes #33512

When using `dbt-fusion` (dbt CLI v2.x) with `dagster-dbt`, dbt tests that successfully complete are incorrectly reported as failed asset checks (`passed=False`) in the Dagster UI.

This happens because `DbtFusionCliEventMessage._get_check_passed()` currently determines success using a strict comparison against `NodeStatus.Success` (`"success"`). While model executions emit `"success"`, dbt tests in `dbt-fusion` emit a `"pass"` status.

Since `"pass" != "success"`, all successful test nodes are incorrectly marked as failed.

This PR updates `_get_check_passed()` to treat both:
- `NodeStatus.Success` (`"success"`) and  
- `TestStatus.Pass` (`"pass"`)

as successful outcomes when evaluating asset checks.

## Impact
- Fixes incorrect failure reporting for dbt test checks in Dagster UI
- Restores parity with `DbtCoreCliEventMessage`, which already handles `TestStatus.Pass` correctly
- Ensures consistent behavior across dbt runtimes (dbt-core and dbt-fusion)

## How I Tested These Changes
- Reviewed the existing `DbtCoreCliEventMessage` pipeline to confirm that `TestStatus.Pass` is treated as a success case.
- Created simulated `dbt-fusion` `NodeFinished` JSON payloads for:
  - `"pass"` (test success)
  - `"fail"` (test failure)
  - `"success"` (model execution)
- Verified that `msg._get_check_passed()` now evaluates to:
  - `True` for `"pass"`
  - `False` for `"fail"`
  - `True` for `"success"`
- Confirmed no changes to other execution paths or dependencies.

### CODE REVIEW

File: python_modules/libraries/dagster-dbt/dagster_dbt/core/dbt_cli_event.py [hunk 1]
```diff
@@ -605,7 +605,7 @@ def is_result_event(self) -> bool:
         return self.raw_event["info"]["name"] == "NodeFinished"
 
     def _get_check_passed(self) -> bool:
-        return self._get_node_status() == NodeStatus.Success
+        return self._get_node_status() in (NodeStatus.Success, TestStatus.Pass)
```
[the remark this answers is not in the document]
Comment (vidiyala99) [reply python_modules/libraries/dagster-dbt/dagster_dbt/core/dbt_cli_event.py]: I’ve added a set of pytest unit tests to explicitly validate the behavior of DbtFusionCliEventMessage._get_check_passed() using mocked dbt-fusion payloads.

The tests cover the key edge cases to prevent regressions:

"pass" (successful test) → True

"fail" (failed test) → False

"success" (model execution) → True

This ensures consistent and correct evaluation across both dbt-core and dbt-fusion runtimes.

**Comment by vidiyala99** [comment]
@OwenKephart mind taking a look when you have a chance? This is a small dagster-dbt / dbt-fusion fix with regression tests.
